pkg/blockbuilder/scheduler/scheduler.go

```diff
@@ -243,6 +250,23 @@ func (s *BlockScheduler) HandleCompleteJob(ctx context.Context, job *types.Job,
 		); err == nil {
 			s.queue.MarkComplete(job.ID(), types.JobStatusComplete)
 			level.Info(logger).Log("msg", "job completed successfully")
+
+			// TODO(owen-d): cleaner way to enqueue next job for this partition,
+			// don't make it part of the response cycle to job completion, etc.
+			jobs, err := s.planner.Plan(ctx, 1)
+			if err != nil {
+				level.Error(logger).Log("msg", "failed to plan subsequent jobs", "err", err)
+			}
+
+			// find first job for this partition
+			nextJob := sort.Search(len(jobs), func(i int) bool {
+				return jobs[i].Job.Partition() >= job.Partition()
+			})
+
+			if nextJob < len(jobs) {
+				_, _, _ = s.idempotentEnqueue(jobs[nextJob])
```

Comment:
there is a potential race here that could result in this operation to fail as we try to enqueue the same job from two different routines (from `HandleCompleteJob` or from the scheduler run loop). but this might not cause any problems

Reply:
This should be ok since the enqueue code first checks the statusmap for existence of a job before adding it to the queue.